mav_trajectory_generation/src/segment.cpp

```diff
@@ -196,4 +196,59 @@ bool Segment::selectMinMaxMagnitudeFromCandidates(
   return true;
 }
 
+bool Segment::getSegmentWithSingleDimension(int dimension,
+                                            Segment* new_segment) const {
+  if (dimension < 0 || dimension >= D_) {
+    LOG(WARNING)
+        << "You shan't ask for a dimension that does not exist in the segment.";
+    return false;
+  }
+
+  *new_segment = Segment(N_, 1);
+  (*new_segment)[0] = polynomials_[dimension];
+  new_segment->setTime(time_);
+  return true;
+}
+
+bool Segment::getSegmentWithAppendedDimension(const Segment& segment_to_append,
+                                              Segment* new_segment) const {
+if (time_ != segment_to_append.getTime()) {
+  LOG(WARNING) << "The appended segment needs the same segment time.";
+  return false;
+}
+if (N_ == 0 || D_ == 0) {
+  *new_segment = segment_to_append;
+  return true;
+}
+if (segment_to_append.N() == 0 || segment_to_append.D() == 0) {
+  *new_segment = *this;
+  return true;
+}
+
+// Get common polynomial order.
+const int N_common = std::max(segment_to_append.N(), N_);
```

Comment:
I wonder if we can have a cheaper check that if N() == N_ we just copy the object directly. Or at least do that inside the polynomial function.